portal-web/test/functional/com/liferay/portalweb/macros/json/layout/JSONLayoutSetter.macro

```diff
@@ -117,6 +117,13 @@ definition {
 				typeSettings = "${typeSettings}",
 				widgetName = "${widgetName}");
 		}
+		else if (isSet(widgetName) && isSet(remoteAppEntryId)) {
+			var typeSettings = JSONLayoutUtil._addWidgetToTypeSettings(
+				column = "${column}",
+				remoteAppEntryId = "${remoteAppEntryId}",
+				typeSettings = "${typeSettings}",
+				widgetName = "${widgetName}");
+		}
```

Comment:
I don't think you need to create a separate else if statement here if we go with 

> Maybe instead, right after _generateWidget, we can check if (isSet(remoteAppEntryId)) and append the remoteAppEntryID.